lib/openstack_query/docs/developer_docs/ADDING_NEW_QUERIES.md

```diff
@@ -0,0 +1,236 @@
+A new subclass of the `PropEnum` class is required to store valid property enums and mappings for your new openstack
+resource. To do this, you must:
+1. Identify all available properties for the new query.
+   - You can find this information by looking in the openstacksdk api documentation [here](https://docs.openstack.org/openstacksdk/latest/user/index.html#api-documentation).
+2. Create a new file under `/enums/query/props/` called `<resource>_properties.py`, replace `<resource>` with the name of the openstack resource you want to query
+3. Create the `<resource>Properties` class definition
+   - class must inherit from `PropEnum` abstract base class and implement abstract methods.
+
+```python
+class ResourceProperties(PropEnum):
+
+    # add enums for each property you found here like so
+    PROP_1 = auto() # replace `PROP_1` with property name
+    ...
+
+    @staticmethod
+    def get_prop_mapping(prop):
+        ...
+        # add prop mappings here.
+        mapping = {
+           ResourceProperties.PROP_1: lambda a: a["prop_1"]
+        }
+        try:
+           return mapping[prop]
+        except KeyError as exp:
+            raise QueryPropertyMappingError(
+                f"Error: failed to get property mapping, property {prop.name} is not supported in FlavorProperties"
+            ) from exp
+
+    @staticmethod
+    def get_marker_prop_func():
+        ...
+        # marker prop func returns the mapped function for the property that can be used as the marker
+        # for pagination
+        # here you need to replace ResourceProperties.PROP_1 with the appropriate property enum that can be used as the
+        # marker for pagination - this is usually the ID property
+        return ResourceProperties.get_prop_mapping(ResourceProperties.PROP_1)
+```
+
+## 2. Create a Runner Class
+A new subclass of the `RunnerWrapper` class is required to store how to actually run the query using openstacksdk. It
+will also be the place where we can define extra parameters that can be used to fine-tune the query.
+
+To add a Runner Class:
+1. Create a new file under `/openstack_query/runners/` called `<resource>_runner.py` replace `<resource>` with openstack resource name you want to query
+2.  Create the `<resource>Runner` class definition
+   - class must inherit from `RunnerWrapper` abstract base class and implement abstract methods.
+
+for example:
+```python
+class ResourceRunner(RunnerWrapper):
+    ...
+    # specify the openstacksdk resource type
+    RESOURCE_TYPE = Resource
+
+    def _parse_meta_params(
+        self,
+        conn: OpenstackConnection
+
+        # define meta params that could be passed
+        # they should all be optional, if no meta-params are passed, the query should still work with default values
+        meta_param_1 = None,
+        meta_param_2 = None,
+        ...
+    ):
+        # Define logic here that will alter the keyword arguments that will be passed to the openstacksdk command
```

Comment:
Could we point people to an example, `, for example in ServerQuery we modify x based on y`